client/state/first-view/actions.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * External dependencies
+ */
+import moment from 'moment';
+
+/**
+ * Internal dependencies
+ */
+import {
+	FIRST_VIEW_EMBRACE,
+	FIRST_VIEW_DISMISS,
+	FIRST_VIEW_HIDE,
+	FIRST_VIEW_SHOW,
+} from 'state/action-types';
+
+import { FIRST_VIEW_START_DATES } from './constants';
+
+export function init( { user } ) {
+	const userCreated = moment( user.get().date );
+	let eligibleViews = [];
+
+	for ( let view in FIRST_VIEW_START_DATES ) {
+		if ( moment( FIRST_VIEW_START_DATES[ view ] ).isBefore( userCreated ) ) {
+			eligibleViews.push( view );
+		}
+	}
+
+	return {
+		type: FIRST_VIEW_SHOW,
+		views: eligibleViews,
+	};
+}
+
+export function dismissView( { view } ) {
+	return {
+		type: FIRST_VIEW_DISMISS,
+		view,
+	};
+}
+
+export function embraceView( { view } ) {
+	return {
+		type: FIRST_VIEW_EMBRACE,
+		view,
+	};
+}
+
+export function hideViews( { views } ) {
+	return {
+		type: FIRST_VIEW_HIDE,
+		views
+	};
+}
+
+export function showViews( { views, force } ) {
```

Comment:
> Where will init be called from?

Boot module is one place I can think of, e.g.

https://github.com/Automattic/wp-calypso/blob/master/client/boot/index.js#L40
https://github.com/Automattic/wp-calypso/blob/master/client/boot/index.js#L213

> maybe I can accomplish the intent of init by putting the login in DESERIALIZE somehow.

Reducer should be pure, it should react to actions, not generate one. Calling `init` would be a side effect, so it does not feel like a good place for this logic. The reducer should only do calculations in order to return the next state.